cityofaustin/atd-vz-data
# Show level 5 polygons on Location view map

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/11735

This PR:
- adds some error handling so any null polygon shapes will not break the view in the future
- updates the location map to fit the polygon on load
- populates the `shape`, `latitude`, and `longitude` columns of level 5 location records in the `atd_txdot_locations` table  (more on this in the comments in the SQL below)
- sneaks in an update to the GraphQL abstract to skip `order_by` from applying to aggregate queries. I can't think of a case where this would be needed

## Testing
**URL to test:** 
https://deploy-preview-1203--atd-vze-staging.netlify.app

**Steps to test:**
1. Go to any of the following crashes: 
   - https://deploy-preview-1203--atd-vze-staging.netlify.app/#/locations/83405791792 
   - https://deploy-preview-1203--atd-vze-staging.netlify.app/#/locations/79DD65A6780 
   - https://deploy-preview-1203--atd-vze-staging.netlify.app/#/locations/9C95D829B2D
   - https://deploy-preview-1203--atd-vze-staging.netlify.app/#/locations/CAFDB95173F
      - you should see a yellow outline of the shape of this level 5 polygon
      - the map should be centered on the polygon
2. You can compare this to https://visionzero.austin.gov/editor/#/locations/83405791792 which breaks on load.
3. Go to https://deploy-preview-1203--atd-vze-staging.netlify.app/#/locations/C389A2CEFC1
   - I removed the `shape`, `latitude`, and `longitude` from this record
   - it should still load and the map will default to downtown

**Note: The following updates have been applied to the staging database. We will need to make the same changes to production when releasing.**

```sql
-- The UI shows the shape column on the map.
-- Level 1-4 locations have this same value in both shape and geometry columns
-- so we follow suit for level 5 records.
UPDATE
	atd_txdot_locations
SET
	shape = geometry
WHERE
	location_group = 2;
```

```sql
-- The latitude and longitude columns are used to center the map on the polygon
-- and query Nearmap API for a timestamp of the latest images captured for the location tiles.
UPDATE
	atd_txdot_locations
SET
	longitude = ST_X(ST_CENTROID(shape)), 
	latitude  = ST_Y(ST_CENTROID(shape))
WHERE
	location_group = 2;
```

---
#### Ship list
